src/cli/README_SRP_CLIENT.md

```diff
@@ -0,0 +1,300 @@
+# OpenThread CLI - SRP Client
+
+## Quick Start
+
+### Build with CoAP API support
+
+Use the `SRP_CLIENT=1 ECDSA=1` build switches to enable SRP Client API support (which requires ECDSA support).
+
+```bash
+> ./bootstrap
+> make -f examples/Makefile-simulation SRP_CLIENT=1 ECDSA=1
+```
+
+## Command List
+
+Usage : `srp client [command] ...`
+
+- [help](#help)
+- [callback](#callback)
+- [start](#start)
+- [stop](#stop)
+- [leaseinterval](#leaseinterval)
+- [keyleaseinterval](#keyleaseinterval)
+- [host](#host)
+- [service](#service)
+
+## Command Details
+
+### help
+
+Usage: `srp client help`
+
+Print SRP client help menu.
+
+```bash
+> srp client help
+callback
+help
+host
+keyleaseinterval
+leaseinterval
+service
+start
+stop
+Done
+```
+
+### callback
+
+Usage `srp client callback [enable|disable]`
+
+Enable/Disable printing callback events from SRP client.
+
+Get current callback mode
+
+```bash
+> srp client callback
+Disabled
+Done
+```
+
+Set callback mode
+
+```bash
+> srp client callback enable
+Done
+
+> srp client callback
+Enabled
+Done
+```
+
+#### Example
+
+When two services are successfully registered:
+
+```bash
+SRP client callback - error:OK
+Host info:
+    name:"dev4312", state:Registered, addrs:[fd00:0:0:0:0:0:0:1]
+Service list:
+    instance:"ins2", name:"_test2._udp", state:Registered, port:111, priority:1, weight:1
+    instance:"ins1", name:"_test1._udp", state:Registered, port:777, priority:0, weight:0
+```
+
+When service `ins2` is removed:
+
+```bash
+SRP client callback - error:OK
+Host info:
+    name:"dev4312", state:Registered, addrs:[fd00:0:0:0:0:0:0:1]
+Service list:
+    instance:"ins1", name:"_test1._udp", state:Registered, port:777, priority:0, weight:0
+Removed service list:
+    instance:"ins2", name:"_test2._udp", state:Removed, port:111, priority:1, weight:1
+```
+
+When host info (and all services) is removed:
+
+```bash
+SRP client callback - error:OK
+Host info:
+    name:"dev4312", state:Removed, addrs:[fd00:0:0:0:0:0:0:1]
+Service list:
+Removed service list:
+    instance:"ins1", name:"_test1._udp", state:Removed, port:777, priority:0, weight:0
+```
+
+### start
+
+Usage: `srp client start <serveraddr> <serverport>`
+
+Start the SRP client with a given server IPv6 address and port number
+
+```bash
+> srp client start fd00::d88a:618b:384d:e760   4724
+Done
+```
+
+### stop
+
+Usage: `srp client stop`
+
+Stop the SRP client.
+
+```bash
+> srp client stop
+Done
+```
+
+### leaseinterval
+
+Usage: `srp client leaseinterval [interval]`
+
+Get the lease interval (in seconds).
+
+```bash
+> srp client leaseinterval
+7200
+Done
+>
+```
+
+Set the lease interval.
+
+```bash
+> srp client leaseinterval 3600
+Done
+```
+
+### keyleaseinterval
+
+Usage: `srp client keyleaseinterval [interval]`
+
+Get the key lease interval (in seconds).
+
+```bash
+> srp client keyleaseinterval
+1209600
```

Comment:
Out of curiosity, any significance to the 14 days?

Reply:
It is what the SRP spec currently suggests a reasonable default value. 

>  A default  limit of two hours for the Lease and 14 days for the SIG(0) KEY are currently thought to be good choices.